### ISSUE TITLE
Bento Unit Test Isolation

### ISSUE THREAD

**Issue by samouri** [ampproject/amphtml#36239]
### Description

I believe Bento unit tests are not properly isolated from each other. In the below example, the effects from the `amp-iframe` run during the tests of `amp-render`, causing a thrown error:

```
FAILED TESTS:
  amp-render-v1.0
    ● "before each" hook for "renders from amp-state"
      Chrome Headless 94.0.4606.61 (Mac OS 10.15.7)
    Error: Uncaught TypeError: Cannot read properties of null (reading 'addEventListener') (/var/folders/zb/8dhhvh5n5c51pntk5ftm1vjh00kpsq/T/9914e7c3fd7c7098234e6fe35982712c-bundle.js:46041)
```
---

We should ensure that a test has fully been cleaned up and all of its effect hooks have completed running before moving on to the next

### Reproduction Steps

```
npx amp unit --headless --files=extensions/amp-iframe/1.0/test/test-amp-iframe.js,extensions/amp-render/1.0/test/test-amp-render.js
```

**Comment by rsimha** [comment]
Here is the cleanup code we currently run after each unit and integration test. Perhaps more needs to be done to satisfy the bento use case?

https://github.com/ampproject/amphtml/blob/f23e77d1428a64c5039887fd05f3e73581f506cf/testing/init-tests.js#L127-L189

**Comment by samouri** [comment]
Yes. We need to ensure a Bento component's effect hooks have all run before advancing to the next test. That said, I don't think this functionality is exposed by Preact. Currently brainstorming how to achieve the same result.
